lib/puma/server.rb

```diff
@@ -478,6 +479,16 @@ def process_client(client, buffer)
       end
     end
 
+    # Compute maximum time to wait for another request arriving on
+    # a client when keep-alive is enabled.
+    # If the pool we created has a maximum of one worker thread we
+    # can't afford to let it wait for the next request while other
+    # work might be pending.
+    def fast_track_ka_timeout
+      Integer(@max_threads) > 1 ? FAST_TRACK_KA_TIMEOUT : 0.0
```

Comment:
You're right. In a previous Puma version that I tried, `@max_threads` was a String at that point, but it's no longer the case.